playbooks/maas-tigkstack-telegraf.yml

```diff
@@ -0,0 +1,80 @@
+---
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Deploy telegraf
+  hosts: "hosts:all_containers"
+  gather_facts: true
+  tasks:
+    - name: Add influxdata apt-keys
+      apt_key:
+        url: "https://repos.influxdata.com/influxdb.key"
+        state: "present"
+
+    - name: Add influxdata repo
+      apt_repository:
+        repo: "deb https://repos.influxdata.com/{{ ansible_distribution | lower }} {{ ansible_distribution_release }} stable"
+        state: "present"
+        update_cache: yes
+
+    - name: Install telegraf
+      apt:
+        pkg: "telegraf"
+        state: "latest"
+
+    - name: Create telegraf plugin dir
+      file:
+        path: "/opt/telegraf"
+        state: directory
+        mode: "0755"
+
+    - name: Return Commands for metric
+      command: "/usr/lib/rackspace-monitoring-agent/plugins/maas_telegraf_format.py"
+      changed_when: false
+      register: telegraf_maas_commands
+      when:
+        - inventory_hostname in groups['hosts']
+
+    - name: Set the telegraf maas commands
+      set_fact:
+        telegraf_commands: "{{ telegraf_maas_commands.stdout_lines }}"
+      when:
+        - inventory_hostname in groups['hosts']
+
+    - name: Drop telegraf config file
+      template:
+        src: templates/tigkstack/telegraf.conf.j2
+        dest: /etc/telegraf/telegraf.conf
+      register: telegraf_config
+
+    - name: Set telegraf defaults
+      copy:
+        content: |
+          USER="root"
+          GROUP="root"
+        dest: /etc/default/telegraf
```

Comment:
yes. based on what I've been able to ascertain this works, however, I need to build some tests to validate.